### PR TITLE
Masking dev

### PR DESCRIPTION
**Pull request by Sh-imaa**
- Conf files
- Few edits to notebooks (those maybe be removed partially after everything is settled)
- reconciled the scripts to fit the old code 
- .eval and .train depending on the code
- fixed dimension bug in one-hot-encoding
- shuffling
- add test to loader

### CODE REVIEW

File: glacier_mapping/train.py [hunk 1]
```diff
@@ -40,7 +40,9 @@ def train_epoch(loader, frame, metrics_opts):
       and the metrics on the training set.
     """
     loss, metrics = 0, {}
+    frame.model.train()
     for x, y in loader:
+        frame.optimizer.zero_grad()
```
Comment (krisrs1128) [inline glacier_mapping/train.py]: Why is this line necessary? Isn't zero_grad called in frame.optimize()?

File: scripts/train.py [hunk 2]
```diff
@@ -31,30 +42,53 @@
...
+    loaders = fetch_loaders(data_dir, args.batch_size, shuffle=True)
+
+    # TODO:handle this error better
+    # if input mask dimension different than outchannels
+    outchannels = conf.model_opts.args.outchannels
+    y_channels = [y.shape[-1] for _, y in loaders["val"]][0]
+    if y_channels != outchannels:
+        raise ValueError("Output dimension is different from model outchannels.")
+
+    # TODO: try to have less nested if/else
```
Comment (krisrs1128) [inline scripts/train.py:54]: I think it would be clearer to just hard code loss_fn to the value on line 60 here. This is just a script, not a package function, so it doesn't need to be entirely general. If people want to try many different loss functions, we could create a package helper to fetch specific cases, but until that need arises, I don't think we need to write code to handle it.

**Review by krisrs1128 — COMMENTED** [review]
I think other than the two small comments everything can be merged as is.

File: glacier_mapping/train.py (hunk 1 above)
Comment (Sh-imaa) [reply glacier_mapping/train.py]: You're right. I changed it with mode.train and missed that we call it already in frame.optimize()

File: scripts/train.py (hunk 2 above)
Comment (Sh-imaa) [reply scripts/train.py:54]: I agree with that we don't need much generality in this script. This helps, for now, with the error if someone uses different channels than specified in the config file. This error will run and won't be caught due to some size manipulation in the rest of the library. It also has occurred with us, so I assume it can occur to another user easily.  
We can move it to a more suitable place though. It was an ad-hoc solution to a problem that happen to multiple people.

Comment (krisrs1128) [reply scripts/train.py:54]: I see, if the check is needed for current code to work, there's no need to refactor it now. You can merge it whenever you would like.
